src/Components/Web.JS/src/Platform/SsrStartOptions.ts

```diff
@@ -7,4 +7,10 @@ export interface SsrStartOptions {
    * (for example, during enhanced navigation or streaming rendering).
    */
   disableDomPreservation?: boolean;
+
+  /**
+   * Configures how long to wait after all Blazor Server components have been removed from the document
+   * before closing the circuit.
+   */
+  circuitInactivityTimeoutMs?: number;
```

Comment:
We _could_ decide to dispose the circuit as soon as we know the page doesn't contain any server components. However, that can cause async component disposal logic to fail if it involves cleaning up browser-side state. This timeout decides how long components are allowed to take to dispose before the circuit closes. The default is 2 seconds, but that was chosen arbitrarily. We could easily make the default longer or shorter.

Reply:
This is good. One situation we definitely want to avoid is navigating from one interactive server page to another causing the circuit to be terminated and instantly restarted for the new page. 2 seconds is reasonable.